common/persistence/execution_manager_test.go

```diff
@@ -0,0 +1,240 @@
+			},
+		},
+		{
+			method: "DeleteWorkflowExecution",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().DeleteWorkflowExecution(gomock.Any(), gomock.Any()).Return(nil)
+			},
+		},
+		{
+			method: "DeleteCurrentWorkflowExecution",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().DeleteCurrentWorkflowExecution(gomock.Any(), gomock.Any()).Return(nil)
+			},
+		},
+		{
+			method: "GetCurrentExecution",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().GetCurrentExecution(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+		{
+			method: "ListCurrentExecutions",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().ListCurrentExecutions(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+		{
+			method: "IsWorkflowExecutionExists",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().IsWorkflowExecutionExists(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+		{
+			method: "GetReplicationDLQSize",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().GetReplicationDLQSize(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+	} {
+		t.Run(tc.method, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			mockedStore := NewMockExecutionStore(ctrl)
+			tc.prepareMocks(mockedStore)
+			manager := NewExecutionManagerImpl(mockedStore, testlogger.New(t), nil)
+			v := reflect.ValueOf(manager)
+			method := v.MethodByName(tc.method)
+			methodType := method.Type()
+			args := methodType.NumIn()
+			var vals []reflect.Value
+			// If a method requires arguments, we expect the first argument to be a context
+			// and the rest to be zero values of the correct type.
+			// For methods like Close and GetShardID, we don't expect any arguments.
+			if args > 0 {
+				vals = append(vals, reflect.ValueOf(context.Background()))
+				for i := 1; i < args; i++ {
+					vals = append(vals, reflect.Zero(methodType.In(i)))
+				}
+			}
+
+			callRes := method.Call(vals)
+			if callRes == nil {
+				return
+			}
+			resultErr := callRes[len(callRes)-1].Interface()
+			err, ok := resultErr.(error)
+			if ok {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestGetReplicationTasks(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockedStore := NewMockExecutionStore(ctrl)
+	mockedStore.EXPECT().GetReplicationTasks(gomock.Any(), gomock.Any()).Return(&InternalGetReplicationTasksResponse{
+		Tasks: []*InternalReplicationTaskInfo{
+			{
+				DomainID: "test",
+				TaskID:   1,
+			},
+			{
+				DomainID: "test",
+				TaskID:   2,
+			},
+		},
+		NextPageToken: nil,
+	}, nil)
+
+	manager := NewExecutionManagerImpl(mockedStore, testlogger.New(t), nil)
+	res, err := manager.GetReplicationTasks(context.Background(), &GetReplicationTasksRequest{})
```

Comment:
could be turned into table test to also cover the [err case](https://github.com/uber/cadence/blob/d60041fffc239ee2f2392a67fedee528fc45512d/common/persistence/executionManager.go#L887)